all/HOPE-3--Dependency-Injection.md

```diff
@@ -78,45 +78,52 @@ It should also be possible to specify different dependency providers for differe
 
 ```python
     # Multiple interfaces using one provider
-    @hug.provides("mysql", interfaces=['http', 'websocket'])
+    @hug.provide("mysql", interfaces=['http', 'websocket'])
     def mysql_web():
         ...
 
 
     # For a provider for a single interface, this shorthand can be used
-    @hug.cli.provides("mysql")
+    @hug.cli.provide("mysql")
     def mysql_cli():
         ...
 
 
     # The following would also be a valid way to apply to multiple interfaces
-    @hug.http.provides("mysql")
-    @hug.cli.provides("mysql")
+    @hug.http.provide("mysql")
+    @hug.cli.provide("mysql")
+    def mysql():
+        ...
+
+    # Or the following would also be a valid way to apply to multiple interfaces
+    @hug.http.provide
+    @hug.cli.provide
     def mysql():
         ...
 ```
 
 If you only want to enable provider usage locally, and don't want to allow it to be overridden, you can explicitly set it to stay unnamed:
 
 ```python
-    @hug.provides(name=False)
+    @hug.provide(name=False)
```

Comment:
Is `name` necessary in addition to the `api` kwarg? You could maybe pass `__name__` to API instead.

Just thinking of opportunities to simplify the interface for testing.